api/test/integration/conftest.py

```diff
@@ -326,86 +497,40 @@ def save_logs(test_name: str):
-    """This function is responsible for setting up the Docker environment necessary for every test.
-    This function will be executed with all the integrated API tests.
+    """Build the image if the parameter nobuild is not used.
+    The function is executed before all the tests are run.
 
     Parameters
     ----------
     request : _pytest.fixtures.SubRequest
         Object that contains information about the current test
     """
 
-    def clean_up_env():
-        """Clean temporary folder, save environment logs and status; and stop and remove all Docker containers."""
-        clean_tmp_folder()
-        if request.session.testsfailed > 0:
-            save_logs(f"{rbac_mode}_{module.split('.')[0]}" if rbac_mode else f"{module.split('.')[0]}")
-
-        # Get the environment current status
-        global environment_status
-        environment_status = get_health()
-        down_env()
-
     # Get the value of the mark indicating the test mode. This value will vary between 'cluster' or 'standalone'
     mode = request.node.config.getoption("-m")
-    env_mode = standalone_env_mode if mode == 'standalone' else cluster_env_mode
-
-    # Add clean_up_env as fixture finalizer
-    request.addfinalizer(clean_up_env)
-
-    test_filename = request.node.config.args[0].split('_')
-    if 'rbac' in test_filename:
-        rbac_mode = test_filename[2]
-        module = test_filename[3]
-    else:
-        rbac_mode = None
-        module = test_filename[1]
-
-    clean_tmp_folder()
-
-    if rbac_mode:
-        change_rbac_mode(rbac_mode)
-        rbac_custom_config_generator(module, rbac_mode)
-    else:
-        enable_white_mode()
-
-    general_procedure(module)
-    values = build_and_up(interval=10, build=request.config.getoption('--nobuild'), env_mode=env_mode)
-
-    while values['retries'] < values['max_retries']:
-        managers_health = check_health(interval=values['interval'],
-                                       only_check_master_health=env_mode == standalone_env_mode)
-        agents_health = check_health(interval=values['interval'], node_type='agent', agents=list(range(1, 9)))
-        nginx_health = check_health(interval=values['interval'], node_type='nginx-lb')
-        # Check if entrypoint was successful
-        try:
-            error_message = subprocess.check_output(["docker", "exec", "-t", "env_wazuh-master_1", "sh", "-c",
-                                                     "cat /entrypoint_error"]).decode().strip()
-            pytest.fail(error_message)
-        except subprocess.CalledProcessError:
-            pass
-
-        if managers_health and agents_health and nginx_health:
-            time.sleep(values['interval'])
-            return
-        else:
-            values['retries'] += 1
+    global env_mode
+    env_mode = mode if mode == standalone_env_mode else cluster_env_mode
+    
+    global build
+    build = request.config.getoption('--nobuild')
+    if build:
+        build_images()
```

Comment:
Question here: I see that the `build` variable is used only in this function but it has a global scope, is it being used anywhere else?